datanode/sqlstore/orders.go

```diff
@@ -98,6 +99,39 @@ func (os *Orders) GetOrder(ctx context.Context, orderIDStr string, version *int3
 	return order, os.wrapE(err)
 }
 
+// GetByMarketAndID returns all orders with given IDs for a market.
+func (os *Orders) GetByMarketAndID(ctx context.Context, marketIDstr string, orderIDs []string) ([]entities.Order, error) {
+	if len(orderIDs) == 0 {
+		os.log.Warn("GetByMarketAndID called with an empty order slice",
+			logging.String("market ID", marketIDstr),
+		)
+		return nil, nil
+	}
+	defer metrics.StartSQLQuery("Orders", "GetByMarketAndID")()
+	marketID := entities.MarketID(marketIDstr)
+	// IDs := make([]entities.OrderID, 0, len(orderIDs))
+	IDs := make([]interface{}, 0, len(orderIDs))
+	in := make([]string, 0, len(orderIDs))
+	bindNum := 2
+	for _, o := range orderIDs {
+		IDs = append(IDs, entities.OrderID(o))
+		in = append(in, fmt.Sprintf("$%d", bindNum))
+		bindNum++
+	}
+	bind := make([]interface{}, 0, len(in)+1)
+	// set all bind vars
+	bind = append(bind, marketID)
+	bind = append(bind, IDs...)
+	// select directly from orders_live table, the current view searches in orders
+	// this is used to expire orders, which have to be, by definition, live. This table uses ID as its PK
+	// so this is a more optimal way of querying the data.
```

Comment:
nitpick: This is slightly at-odds with the name of the function, which doesn't mention anything about expiring or whatnot.. I guess if the function were called GetLiveOrdersByMarketAndID it would be a bit more obvious but it is a bit of a mouthful.. it's perhaps slightly surprising to a user that two functions:
`GetByMarket` and `GetByMarketAndID` are actually querying different things